tests/manager-test.js

```diff
@@ -1,15 +1,97 @@
 import chai from 'chai';
 const expect = chai.expect;
-import Manager from '../manager.js';
+import Manager from '../classes/manager.js';
+import Hotel from '../classes/hotel.js';
+import Booking from '../classes/booking.js';
+
+let roomData = [{
+  number: 1,
+  roomType: "residential suite",
+  bidet: true,
+  bedSize: "queen",
+  numBeds: 1,
+  costPerNight: 358.4
+},
+{
+  number: 2,
+  roomType: "suite",
+  bidet: false,
+  bedSize: "full",
+  numBeds: 2,
+  costPerNight: 477.38
+},
+{
+  number: 3,
+  roomType: "single room",
+  bidet: false,
+  bedSize: "king",
+  numBeds: 1,
+  costPerNight: 491.14
+},
+{
+  number: 4,
+  roomType: "single room",
+  bidet: false,
+  bedSize: "queen",
+  numBeds: 1,
+  costPerNight: 429.44
+}];
+let bookingData = [{
+  id: "5fwrgu4i7k55hl6sz",
+  userID: 1,
+  date: "2020/01/10",
+  roomNumber: 1,
+  roomServiceCharges: []
+},
+{
+  id: "5fwrgu4i7k55hl6t5",
+  userID: 43,
+  date: "2020/01/24",
+  roomNumber: 2,
+  roomServiceCharges: []
+},
+{
+  id: "5fwrgu4i7k55hl6t6",
+  userID: 1,
+  date: "2020/01/10",
+  roomNumber: 3,
+  roomServiceCharges: []
+},
+{
+  id: "5fwrgu4i7k55hl6t7",
+  userID: 20,
+  date: "2020/02/16",
+  roomNumber: 7,
+  roomServiceCharges: []
+}]
+let mngData = { id: 51, name: 'mr.manager' };
 
 describe('Manager', function () {
-  let manager;
+  let manager, bookings, hotel;
 
   beforeEach(() => {
-    manager = new Manager();
+    bookings = bookingData.map(booking => new Booking(booking));
+    hotel = new Hotel('Grandiose Sarajevo Motel', roomData, bookings);
+    manager = new Manager(mngData);
   });
 
   it('should be a function', () => {
     expect(Manager).to.be.a('function');
   });
+
+  it('should have an ID', () => {
+    expect(manager.id).to.equal(51);
+  });
+
+  it('should have a name', () => {
+    expect(manager.name).to.equal('mr.manager');
+  });
+
+  it('should be able to calculate percentage of rooms booked', () => {
+    expect(manager.calcPercRoomsBooked(hotel, '2020/01/24')).to.deep.equal(.25);
```

Comment:
Your test look great! Cool little trick I learned is that 'eql' is equivalent to 'deep.equal'

Reply:
oooooh, that's handy, thanks!